src/main/java/biz/wgc/fwplan/tools/PDFCreator.java

```diff
@@ -0,0 +1,196 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.common.PDRectangle;
+import org.apache.pdfbox.pdmodel.font.PDType1Font;
+import org.apache.pdfbox.pdmodel.graphics.image.PDImageXObject;
+
+import com.itextpdf.html2pdf.ConverterProperties;
+import com.itextpdf.html2pdf.HtmlConverter;
+import com.itextpdf.html2pdf.attach.impl.DefaultTagWorkerFactory;
+import com.itextpdf.kernel.pdf.PdfDocument;
+import com.itextpdf.kernel.pdf.PdfString;
+import com.itextpdf.kernel.pdf.PdfViewerPreferences;
+import com.itextpdf.kernel.pdf.PdfWriter;
+import com.itextpdf.kernel.pdf.WriterProperties;
+import com.itextpdf.layout.font.FontProvider;
+
+import biz.wgc.fwplan.constants.ValueHolder;
+import biz.wgc.fwplan.data.Plan;
+import biz.wgc.fwplan.data.UserElement;
+import biz.wgc.fwplan.data.db.DBUserElement;
+import biz.wgc.fwplan.helper.ImagePaint;
+
+public class PDFCreator {
+	private static Logger logger = Logger.getLogger(PDFCreator.class);
+
+	private PDFCreator() {
+	}
+
+	public static void writePDF() {
+		try {
+
+			PDDocument document = new PDDocument();
+			document.addPage(createMapPage(document));
+			createTextPage();
+			document.save(ValueHolder.INSTANCE.getOutputPath() + "plan-"
+					+ ValueHolder.INSTANCE.getPlan().getPlanNumber() + "_Site2.pdf");
+			document.close();
+
+		} catch (Exception e) {
+			logger.error("Error in create PDF ", e);
+		}
+	}
+
+	private static PDPage createMapPage(PDDocument document) throws IOException {
+		PDPage page = new PDPage(new PDRectangle(PDRectangle.A3.getHeight(), PDRectangle.A3.getWidth()));
+		List<UserElement> userElements = DBUserElement.getInstance()
+				.getAllElementsForPlan(ValueHolder.INSTANCE.getPlan().getId());
+		File printFile = ImagePaint.createImageFile(userElements);
+		PDImageXObject pdImage = PDImageXObject.createFromFile(printFile.getAbsolutePath(), document);
+		PDPageContentStream contentStream = new PDPageContentStream(document, page);
+		contentStream.drawImage(pdImage, 0, 0);
+		contentStream.beginText();
+		contentStream.newLineAtOffset(25, PDRectangle.A3.getWidth() - 25);
+		contentStream.setFont(PDType1Font.TIMES_ROMAN, 20);
+		contentStream.showText("Einsatzplan " + ValueHolder.INSTANCE.getPlan().getPlanNumber());
+		contentStream.endText();
+		contentStream.close();
+		return page;
+	}
+
+	private static void createTextPage() throws IOException {
+		String path = ValueHolder.INSTANCE.getOutputPath() + "plan-" + ValueHolder.INSTANCE.getPlan().getPlanNumber()
+				+ "_Site1.pdf";
+		FileOutputStream outputStream = new FileOutputStream(path);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *outputStream*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by WriterProperties constructor. Either a) close *outputStream* in a try-finally block or b) close the resource by declaring *outputStream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).